File: leetcode/119.pascals-triangle-ii.py
class Solution(object):
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        row = [0 for i in range(rowIndex + 1)]
        
        for i in range(rowIndex + 1):
            for j in range(i, -1, -1):
                if j == 0 or j == i:
                    row[j] = 1
                else:
                    row[j] = row[j - 1] + row[j]

        return row
        

File: leetcode/119.pascals-triangle-ii.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        
        vector<int> row(rowIndex + 1, 0);

        for (auto i = 0; i < rowIndex + 1; ++i)
        {
            for (auto j = i; j >= 0; --j)
            {
                if (j == 0 || j == i)
                {
                    row[j] = 1;
                }
                else
                {
                    row[j] = row[j] + row[j - 1];
                }
            }
        }

        return row;
    }
};
